model/infinispan/src/main/java/org/keycloak/spi/infinispan/impl/remote/DefaultCacheRemoteConfigProviderFactory.java

```diff
@@ -357,4 +354,17 @@ private static void addTlsConfig(ProviderConfigurationBuilder builder) {
                 .type(ProviderConfigProperty.STRING_TYPE)
                 .add();
     }
+
+    private static void addCreateRemoteCachesConfig(ProviderConfigurationBuilder builder) {
+        builder.property()
+                .name(CREATE_CACHES)
+                .helpText("If true, Keycloak will create the caches in the external Infinispan cluster.")
+                .type(ProviderConfigProperty.BOOLEAN_TYPE)
+                .add();
```

Comment:
This SPI option replaces the system properties, but I'm having doubts about its use. It is enabled by default, and I'm thinking about removing it. If disabled, Keycloak will fail to start when the caches do not exist.

IMO, it is a useless option. May I remove it? 

@ahus1 @ryanemerson

Reply:
My understanding from past discussions is that if the cache already exists on the Infinispan server, then the cache config we try to apply from Keycloak is ignored as `getOrCreate` is used on the server-side. So I'm happy removing the option altogether as it is one less thing to potentially confuse users.

Reply:
The one situation where it makes sense to fail the startup when the caches do not exist, is when you want to have full control over the cache configurations, and want KC to prevent the automatic setup in case your manual setup fails / leads to any error. 

I could also imagine a race condition where ISPN started, but the Operator did not yet create the caches. And in exactly that moment Keycloak starts, creates the caches, and then the Operator can't apply the custom configuration. 

So I'd say this is valid for some setups, still I'd say having a new default that creates the caches automatically is the way for the future. 

If you still consider this option useless, feel free to remove it.

Reply:
> I could also imagine a race condition where ISPN started, but the Operator did not yet create the caches. And in exactly that moment Keycloak starts, creates the caches, and then the Operator can't apply the custom configuration.

Just an FYI. The Cache CRs have a "Ready" condition which is only set to True once the cache configuration has been applied on the server. So I would expect/hope that any automated deployment would wait on this condition before proceeding with the Keycloak deployment, especially considering that users have to do this for the Infinispan CR given that Keycloak will fail to startup if the remote cache is not available.

Reply:
> Does the CacheCR wait for the cluster to form, or do they apply the cache configuration when the first Pod starts?

It waits for the cluster to form with all expected pods in the Ready state and also ensures that all cluster members have the same view, i.e. no split-brain.